spec/java/util/HashSet.main.lsl

```diff
@@ -612,25 +612,25 @@ automaton HashSetAutomaton
 
         var i: int = 0;
         val expectedModCount: int = this.modCount;
-        val visitedKeys: map<Object, Object> = action MAP_NEW();
+        val unseenKeys: map<Object, Object> = action MAP_NEW();
+        action MAP_UNITE_WITH(unseenKeys, this.storage);
```

Comment:
The `MAP_CLONE` action may be helpful here.

Reply:
Fixed;